### PR TITLE
Collected Additions

### PR DESCRIPTION
**Pull request by hulpke**
New features:

- Nicer interface to character table computation
- Slight speedup of character reduction
- Confluent RWS for S10(2)
- Speedup for `MinimalGeneratingSet`
- Fix for missing Conjugacy classes (#4866)

## Text for release notes

An issue leading to an error message when computing conjugacy classes in large permutation groups has been fixed.

### CODE REVIEW

File: lib/ctblfuns.gi [hunk 1]
```diff
@@ -1666,7 +1666,9 @@ InstallMethod( ScalarProduct,
     x2:= ValuesOfClassFunction( x2 );
     scpr:= 0;
     for i in [ 1 .. Length( x1 ) ] do
-      scpr:= scpr + x1[i] * GaloisCyc( x2[i], -1 ) * weight[i];
+      if x1[i]<>0 and x2[i]<>0 then
+	scpr:= scpr + x1[i] * GaloisCyc( x2[i], -1 ) * weight[i];
```
Comment (fingolfin) [inline lib/ctblfuns.gi:1670]: looks weird in the webinterface -- uses tabs for indentation?

File: lib/ctblfuns.gi [hunk 2]
```diff
@@ -3269,18 +3271,26 @@ InstallMethod( ReducedClassFunctions,
...
+      while j<=upper[i] do
         scpr:= ScalarProduct( ordtbl, single, constituents[j] );
         if IsInt( scpr ) then
           scpr:= Int( scpr / normsquare[j] );
           if scpr <> 0 then
             single:= single - scpr * constituents[j];
+	    if ForAll(single,x->x=0) then
+	      #Print("break at ",j,"\n");
+	      j:=upper[i];
+	    fi;
```
Comment (fingolfin) [inline lib/ctblfuns.gi:3286]: Mixes tabs and spaces for indentation

File: lib/ctblfuns.gi [hunk 3]
```diff
@@ -3269,18 +3271,26 @@ InstallMethod( ReducedClassFunctions,
 
     for i in [ 1 .. Length( reducibles ) ] do
       single:= reducibles[i];
-      for j in [ 1 .. upper[i] ] do
+
+      #for j in [ 1 .. upper[i] ] do
```
Comment (fingolfin) [inline lib/ctblfuns.gi:3275]:
```suggestion
```

**Review by fingolfin — COMMENTED** [review]
Thank you, very nice!

Just a minor remark: one commit introduces some new code that uses tabs for indentation while the surrounding code uses spaces, which results in hard to read code if one does not use the "correct" tab settings...

File: lib/ctblfuns.gi (hunk 1 above)
Comment (hulpke) [reply lib/ctblfuns.gi:1670]: Yes -- my editor (vim) seems to introduce tabs if the file already contains tabs. I have replaced my tabs with spaces.

File: lib/ctblfuns.gi (hunk 2 above)
Comment (hulpke) [reply lib/ctblfuns.gi:3285]: Ditto

File: lib/ctblfuns.gi (hunk 1 above)
Comment (fingolfin) [reply lib/ctblfuns.gi:1670]: Unfortunately I still see tabs in this line (the other two seem to be fixed, though)

**Review by fingolfin — APPROVED** [review]
Thank you!

**Pull request merged**